src/components/Payment.js

```diff
@@ -0,0 +1,69 @@
+import { Component, PropTypes, DOM as dom, createFactory } from 'react';
+
+class Payment extends Component {
+  constructor(props) {
+    super(props);
+    this.calculateBalance = this.calculateBalance.bind(this);
+    this.renderAmountField = this.renderAmountField.bind(this);
+    this.state = {
+      balance: props.startingBalance,
+      amountInputs: []
+    };
+  }
+
+  calculateBalance() {
+    const paid = this.state.amountInputs.reduce((total, input) => {
+      return total + parseFloat(input.value || 0);
+    }, 0);
+
+    this.setState({ balance: this.props.startingBalance - paid });
+  }
+
+  componentDidMount() {
+    const amountInputs = Array.from(this.refs.form.querySelectorAll('input'));
```

Comment:
yeah that makes sense. From what I've seen, using refs seems to be the correct to get input values.